src/Kinto.elm

```diff
@@ -64,22 +65,69 @@ type Endpoint
     | RecordEndpoint BucketName CollectionName RecordId
 
 
+type Filter
+    = Equal String String
+    | Min String String
+    | Max String String
+    | LT String String
+    | GT String String
+    | IN String (List String)
+    | NOT String String
+    | LIKE String String
+    | SINCE String
+    | BEFORE String
+
+
 type alias Body =
     Encode.Value
 
 
-type Request
-    = KintoRequest Config Endpoint Method
-    | KintoRequestWithBody Config Endpoint Method Body
+type alias Client =
+    { baseUrl : String
+    , headers : List ( String, String )
+    }
 
 
-type alias Config =
-    { baseUrl : String
-    , headers : List Http.Header
-    , auth : Auth
+type alias Resource a =
+    { itemEndpoint : String -> Endpoint
+    , listEndpoint : Endpoint
+    , itemDecoder : Decode.Decoder a
+    , listDecoder : Decode.Decoder (List a)
```

Comment:
Yeah, appart from being "more explicit", the `listDecoder` is pretty useless.
I agree with your concerns, especially regarding the `Endpoint`s: there's another concern I had for a long time, and don't know yet how to deal with... you can call `getList` with an item endpoint, it'll fail at runtime while trying to decode a list instead of a single item...

I guess there's still a lot more work left on this API, maybe for a further iteration?